### ISSUE TITLE
v2.0: ambiguous "skipping" status for command task in check mode

### ISSUE THREAD

**Issue by zarnovican** [ansible/ansible#14075]
##### Issue Type:

Feature Idea
(Bug Report?)
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /home/zarnovic/.ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

No changes.
##### Environment:

N/A
##### Summary:

In check mode, "skipping" status is ambiguous for command tasks with `when` condition.

One major use-case for running check mode is to find out what would be applied in normal mode. Command tasks always return "skipping" status in check mode (those that don't have "always_run: yes", to be exact). User cannot tell whether the command is really skipped due to `when` condition not being met or skipped because it is in check mode.

It would be nice to distinguish these two states. One possible solution is to return "changed" status on commands that are not skipped by `when` condition. This would align the behavior with other modules.

Changing it from "skipped" to "changed" may cause heart-attack for sysadmins already used to current convention. I don't mind how it is fixed, as long as those two states are distinguishable.

Workaround: print command's `when:` condition with debug task ("Next command would run: true/false").

In Ansible 1.9, it was distinguishable albeit the output was slightly confusing:

```
TASK: [unconditional command] ************************************************* 
skipping: [localhost]

ok: [localhost]

TASK: [conditional command] *************************************************** 
skipping: [localhost]
```
##### Steps To Reproduce:
1. create playbook
   
   ```
   ---
   - name: 'test1'
     hosts: localhost
     gather_facts: no
   
     tasks:
       - name: 'unconditional command'
         command: /bin/echo foo
   
       - name: 'conditional command'
         command: /bin/echo bar
         when: "{{ condition | default('false') }}"
   ```
2. run playbook in check mode
##### Expected Results:

```
PLAY [test1] *******************************************************************

TASK [unconditional command] ***************************************************
changed: [localhost]

TASK [conditional command] *****************************************************
skipping: [localhost]
```
##### Actual Results:

```
PLAY [test1] *******************************************************************

TASK [unconditional command] ***************************************************
skipping: [localhost]

TASK [conditional command] *****************************************************
skipping: [localhost]
```

Notice, first task would run in normal mode, while the second would be skipped due to `when:`. Both have the same output in check mode.

**Comment by hloeffler** [comment]
command and shell will never run in check_mode unless you use "always_run: True"

**Comment by zarnovican** [comment]
> command and shell will never run in check_mode unless you use "always_run: True"

I know.

The point is: Can you tell if the command _would_ run in normal mode, just by looking at output of check mode ?

**Comment by hloeffler** [comment]
that would be a nice feature
+1

**Comment by SupremeGit** [comment]
This behaviour is goddamned ridiculous. What is the goddamned point of check mode, when you can't even tell what tasks are going to run?

Sometimes the stupidity of Ansible just boggles the mind. 

The current behaviour is an utter disgrace, and turns any use of ansible in a production environment into a game of russian roulette.

Ansible needs to fix this IMMEDIATELY.

**Comment by bcoca** [comment]
dupe of #9508

**Comment by zarnovican** [comment]
For anybody reaching this ticket.. this is not a bug.

What was not obvious to me (or you), is that you need to run check mode with at least one verbosity level enabled to see the difference. This was implemented pre-2.0.

```
$ ansible-playbook foo.yaml -CD 

PLAY [test1] *******************************************************************

TASK [unconditional command] ***************************************************
skipping: [localhost]

TASK [conditional command] *****************************************************
skipping: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=0   
```

```
$ ansible-playbook foo.yaml -CD -v

PLAY [test1] *******************************************************************

TASK [unconditional command] ***************************************************
skipping: [localhost] => {"changed": false, "msg": "remote module does not support check mode", "skipped": true}

TASK [conditional command] *****************************************************
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=0   
```

**Comment by SupremeGit** [comment]
Thanks for the tip. That does go a long way toward alleviating the issue.
Still, I believe it would've a significant improvement if the difference could be seen without -v.
